src/search/utils/QueryStringGenerator.ts

```diff
@@ -56,12 +60,22 @@ const createValueString = (
     return doubleQuote(id);
   }
 
-  // We are testing for term=xref and valuePrefix=any because the
-  // search API expects the valuePrefix to be ommited in this case.
-  // eg xref:foo rather than xref_any:foo
   let valueString = stringValue;
-  if (valuePrefix && !(term === 'xref' && valuePrefix === 'any')) {
-    valueString = `${valuePrefix}-${stringValue}`;
+  if (term === 'xref') {
+    if (!valuePrefix) {
+      throw new Error('valuePrefix not provided in xref query');
+    }
+    // The API will run more quickly when all database entries are
+    // requested by the user if xref-X:* becomes database:X
+    else if (term === 'xref' && stringValue === '*') {
```

Comment:
you're already checking `term === 'xref'` in the block

Reply:
good catch